### PR TITLE
Added Sonar, Findbugs, Checkstyle, Jacoco. Added h2 database dependen…

### PR DESCRIPTION
**Pull request by Tymoteuszauke**
…cy with configuration in application properties.

### CODE REVIEW

**Comment by mhbratek** [comment]
Block master.

File: botrest/src/main/resources/application.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 spring:
   datasource:
-    url: jdbc:mysql://localhost:3306/BlackBookRobotDB
-    username: root
-    password: 123ZOOMzoom890
```
Comment (mhbratek) [inline botrest/src/main/resources/application.yml:5]: :1st_place_medal: 
My idea is to create file `application.yml.sample` then add to gitignore application.yml, later copy `application.yml.sample` and remove `.sample`.

**Review by mhbratek — APPROVED** [review]

File: botrest/src/main/resources/application.yml (hunk 1 above)
Comment (Tymoteuszauke) [reply botrest/src/main/resources/application.yml:5]: It is temporary solution. Later h2 will be used only for integration tests (and tests will be provided with h2 db access properties) and here we will put data to mysql db.
Eventually there will be another application which will be holding all configuration data for all modules.

Comment (agawda) [inline botrest/src/main/resources/application.yml:5]: Is this your FB password as well? :v

File: botrest/pom.xml [hunk 2]
```diff
@@ -25,33 +24,12 @@
...
-		<dependency>
-			<groupId>org.projectlombok</groupId>
-			<artifactId>lombok</artifactId>
-			<optional>true</optional>
-		</dependency>
-		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-test</artifactId>
-			<scope>test</scope>
-		</dependency>
```
Comment (agawda) [inline botrest/pom.xml:54]: 👍

File: pom.xml [hunk 3]
```diff
@@ -23,9 +29,18 @@
 
     <properties>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (agawda) [inline pom.xml:32]: What about target?

**Review by agawda — APPROVED** [review]

**Pull request merged**